modules/random/include/hephaestus/random/random_object_creator.h

```diff
@@ -0,0 +1,208 @@
+// Random boolean creation
+//=================================================================================================
+template <typename T>
+concept IsBoolean = std::is_same_v<T, bool>;
+
+template <IsBoolean T>
+[[nodiscard]] auto random(std::mt19937_64& mt) -> T {
+  std::bernoulli_distribution dist;
+  return dist(mt);
+}
+
+//=================================================================================================
+// Random integer value creation
+//=================================================================================================
+template <typename T>
+concept IsNonBooleanIntegral = std::integral<T> && !std::same_as<T, bool>;
+
+template <IsNonBooleanIntegral T>
+[[nodiscard]] auto random(std::mt19937_64& mt) -> T {
+  std::uniform_int_distribution<T> dist;
+  return dist(mt);
+}
+
+//=================================================================================================
+// Random floating point value creation
+//=================================================================================================
+template <std::floating_point T>
+[[nodiscard]] auto random(std::mt19937_64& mt) -> T {
+  std::uniform_real_distribution<T> dist;
+  return dist(mt);
+}
+
+//=================================================================================================
+// Random enum creation
+//=================================================================================================
+template <typename T>
+concept IsEnum = std::is_enum_v<T>;
+
+template <IsEnum T>
+[[nodiscard]] auto random(std::mt19937_64& mt) -> T {
+  static const auto enum_values = magic_enum::enum_values<T>();
+  std::uniform_int_distribution<size_t> dist(0, enum_values.size() - 1);
+  return enum_values[dist(mt)];  // NOLINT(cppcoreguidelines-pro-bounds-constant-array-index)
+}
+
+//=================================================================================================
+// Random timestamp creation
+//=================================================================================================
+template <typename T>
+concept IsTimestamp = requires {
+  typename T::clock;
+  typename T::duration;
+  requires std::is_same_v<typename T::clock, std::chrono::system_clock> ||
+               std::is_same_v<typename T::clock, std::chrono::steady_clock>;
+  requires std::is_same_v<T, typename std::chrono::time_point<typename T::clock, typename T::duration>>;
+};
+
+namespace internal {
+template <IsTimestamp T, size_t Year>
+[[nodiscard]] constexpr auto createFinalTimestampOfTheYear() -> T {
+  // THe final date of the year is YYYY-12-31.
+  constexpr auto YEAR = std::chrono::year{ Year };
+  constexpr auto FINAL_MONTH = std::chrono::December;
+  constexpr auto FINAL_DAY = std::chrono::year_month_day_last{ YEAR / FINAL_MONTH / std::chrono::last };
+  constexpr auto FINAL_DATE_SYS_DAYS = std::chrono::sys_days{ FINAL_DAY };
+
+  // The final time of the day is 23:59:59.000...
```

Comment:
why not 23:59:59.99999? std::chrono is by default afaik in nanoseconds, so why not put sth like
`std:chrono::hours{24} - std::chrono::nanoseconds{1}`

Reply:
That is a good suggestions.
Regarding `nanoseconds`, this is actually more intricate.
* the C++ standard does not specify the precision
* gcc uses `nanoseconds`, that is why linux default is nanosecond, and that is why we assume nanoseconds
* clang uses `microseconds`
* mcvc uses `microseconds`

I can deduce the `clock::duration` here and apply your patch of 24h - our.